### DISCUSSION TITLE
What are the differences between the different yhats?

### DISCUSSION THREAD

**Discussion by nachi-lieder** [ourownstory/neural_prophet#1108] [General]
lets say I am predicting 5 steps ahead.  I have 5 yhats (yhat1, yhat2 ....)
what does each resemble? for a given "ds" , these are the 5 days ahead predicted?

Why do some predictions have NaN and some dont?

**Comment by judussoari** [comment]
Hi @nachi-lieder, if you specify `n_forecasts=5`, then you are producing 5 step-forecasts. That means, for every timestep (e.g. once in an hour, or once per day, depending on your data resolution), you are predicting the next 5 steps. 
The NaN values occur because of the autoregression. Say you use `n_lags=10`, then you are always predicting the next 5 steps using the previous 10 as an input. For the first 10 steps of your forecast, a prediction is not available as 10 input values are needed to do a 5 step ahead prediction. 
For each timestep predicted, the result of `NeuralProphet.predict()` gives you the forecasts for step 1-5. these are the yhats.
